src/lib/accessibility/A11yBehaviorFactory.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ButtonBehavior,
+  ToggleButtonBehavior,
+  ImageBehavior,
+  InputBehavior,
+  ListBehavior,
+  ListItemBehavior,
+  SelectableListBehavior,
+  SelectableListItemBehavior,
+  MenuItemBehavior,
+  MenuBehavior,
+  DefaultBehavior,
+  TreeBehavior,
+  TreeItemBehavior,
+  GroupBehavior,
+} from './Behaviors/behaviors'
+import { IAccessibilityBehavior } from './interfaces'
+
+export enum A11yBehaviorType {
+  button,
+  toggleButton,
+  image,
+  input,
+  list,
+  listItem,
+  selectableList,
+  selectableListItem,
+  menu,
+  menuItem,
+  tree,
+  treeitem,
+  group,
+}
+
+export class A11yBehaviorFactory {
```

Comment:
I would invert the logic - behavior should be able to register itself to a factory. This way, we would allow others to add behaviors and use them from their code.

Reply:
nit, for readability and insignificant perf reasons:
I'd add these behaviors in a static map like:
```tsx
private static behaviors: Map< A11yBehaviorType, () => IAccessibilityBehavior<{}, {}> > = new Map([
  A11yBehaviorType.button, () => new ButtonBehavior(),
  A11yBehaviorType. toggleButton, () => new ToggleButtonBehavior(),
  // ... and so on
] as [A11yBehaviorType, () => IAccessibilityBehavior<{}, {}>][])
```
and then `createBehavior` method becomes:
```tsx
public static createBehavior(a11yType?: A11yBehaviorType): IAccessibilityBehavior<{}, {}> {
  return this.behaviors.get(a11yType)
    ? this.behaviors.get(a11yType)()
    : new DefaultBehavior()
}
```
also consider adding types for `IAccessibilityBehavior<{}, {}>` if possible

Reply:
Good idea, I'll change that. Thank you!